src/frontend/components/button/lib/create-report-button.js

```diff
@@ -0,0 +1,82 @@
+import { Component } from "component";
+import { validateRequiredFields } from "validation";
+import { logging } from "logging";
+
+/**
+ * CreateReportButtonComponent class to create a report submission button component
+ * @augments Component
+ */
+class CreateReportButtonComponent extends Component {
+  /**
+   * Create a new CreateReportButtonComponent
+   * @param {HTMLButtonElement} element button element
+   */
+  constructor(element) {
+    super(element);
+    this.el = $(element);
+    this.canSubmitRecordForm = false;
+    this.initSubmitReport();
+  }
+
+  /**
+   * Initialise the submit report button
+   */
+  initSubmitReport() {
+    this.el.on('click', (ev) => { this.submitReport(ev) });
+  }
+
+  /**
+   * Check if at least one checkbox is checked
+   * @param {jQuery<HTMLFieldSetElement>} $fieldset fieldset to check
+   * @returns {boolean} true if at least one checkbox is checked
+   */
+  checkForAtLeastOneValue($fieldset) {
```

Comment:
For this, I'm wondering whether it's possible to reuse the code in `src/frontend/js/lib/validation.js` (e.g. the `validateCheckboxGroup` function)

Reply:
No problem, the main reason this is here is I didn't want to "fiddle" with the code that's already working and in place for validation (and it's easier to test here) - I'll look at moving it over.